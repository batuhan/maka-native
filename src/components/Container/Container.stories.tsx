// src/components/Container/Container.stories.tsx

import React from "react";
import { Text } from "react-native";
import { storiesOf } from "@storybook/react-native";

import Container from "./Container";

storiesOf("Container", module).add("Container", () => (
  <Container>
    <Text>Container</Text>
  </Container>
));
