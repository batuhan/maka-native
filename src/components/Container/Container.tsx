import React, { ReactNode } from "react";
import { View } from "react-native";
import styled from "styled-components";

const ContainerView = styled(View)`
  background-color: ${(props: { bgColor?: string }) =>
    props.bgColor || "white"};
`;

interface Props {
  children: ReactNode;
  bgColor?: string | undefined;
}

export default function({ children, bgColor }: Props) {
  return <ContainerView bgColor={bgColor}>{children}</ContainerView>;
}
