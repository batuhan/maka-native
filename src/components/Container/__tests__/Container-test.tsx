/**
 * @jest-environment jsdom
 */

import React from "react";
import { shallow, mount } from "enzyme";
import { TextInput } from "react-native";

import Container from "../Container";

describe("Container", () => {
  const children = <TextInput>Container Text</TextInput>;

  describe("Rendering", () => {
    it("should render given children", () => {
      const component = shallow(<Container>{children}</Container>);
      expect(component.contains(children)).toEqual(true);
    });

    it("should render with given bgColor", () => {
      const component = mount(<Container>{children}</Container>);
      const tree = component
        .children()
        .first()
        .html();
      expect(tree).toContain("view");
      expect(tree).toContain("textinput");
    });
  });
});
