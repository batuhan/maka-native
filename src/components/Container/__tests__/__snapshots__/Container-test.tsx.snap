// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Rendering should match to snapshot 1`] = `
<View>
  <Text>
    Container Children
  </Text>
</View>
`;
